subsys/net/lib/lwm2m/lwm2m_message_handling.c

```diff
@@ -2926,12 +3091,33 @@ int lwm2m_engine_send(struct lwm2m_ctx *ctx, char const *path_list[], uint8_t pa
 	ret = do_send_op(msg, content_format, &lwm2m_path_list);
 	lwm2m_registry_unlock();
 	if (ret < 0) {
+		if (lwm2m_timeseries_data_rebuild(msg, ret)) {
+			goto msg_init;
+		}
+
 		LOG_ERR("Send (err:%d)", ret);
 		goto cleanup;
 	}
+#if defined(CONFIG_LWM2M_RESOURCE_DATA_CACHE_SUPPORT)
+	msg->cache_info = NULL;
+#endif
 	LOG_INF("Send op to server (/dp)");
 	lwm2m_information_interface_send(msg);
 
+#if defined(CONFIG_LWM2M_RESOURCE_DATA_CACHE_SUPPORT)
+	if (cache_temp_info.entry_size) {
+		/* Init Path list for continuous message allocation */
+		lwm2m_engine_path_list_init(&lwm2m_path_list, &lwm2m_path_free_list,
+					    lwm2m_path_list_buf,
+					    CONFIG_LWM2M_COMPOSITE_PATH_LIST_SIZE);
+
+		if (init_next_pending_timeseries_data(&cache_temp_info, &lwm2m_path_list,
+						     &lwm2m_path_free_list)) {
+			goto msg_alloc;
```

Comment:
Same story here, this goto logic could use some explaination.

Reply:
Ok . I will add comment's.